src/coord/polar/polarCreator.ts

```diff
@@ -102,9 +102,15 @@ function updatePolarScale(this: Polar, ecModel: GlobalModel, api: ExtensionAPI)
     // Fix extent of category angle axis
     if (angleAxis.type === 'category' && !angleAxis.onBand) {
         const extent = angleAxis.getExtent();
-        const diff = 360 / (angleAxis.scale as OrdinalScale).count();
```

Comment:
We can see that the old code takes `360` as the range of `endAngle` to `startAngle`, which is not always the case.